src/services/user/autoFollowUserAnon.js

```diff
@@ -0,0 +1,67 @@
+const {successResponse} = require('../../utils');
+const {v4: uuidv4} = require('uuid');
+const {UserFollowUser, UserFollowUserHistory} = require('../../databases/models');
+const UsersFunction = require('../../databases/functions/users');
+
+const autoFollowUserAnon = async (req, res) => {
+  try {
+    const userFollowUserList = await UserFollowUser.findAll({
+      where: {
+        is_anonymous: false
+      }
+    });
+
+    const userFollowerAnonData = [];
+    const userFollowerAnonDataHistory = [];
+    await Promise.all(
+      userFollowUserList.map(async (followUser) => {
+        const anonId = await UsersFunction.findAnonymousUserId(followUser.user_id_followed);
+
+        const [followAnonUser, isUnfollowByUser] = await Promise.all([
+          UserFollowUser.findOne({
+            where: {
+              user_id_follower: followUser.user_id_follower,
+              user_id_followed: anonId,
+              is_anonymous: true
+            }
+          }),
+          UserFollowUserHistory.findOne({
+            where: {
+              user_id_follower: followUser.user_id_follower,
+              user_id_followed: anonId,
+              action: 'out'
+            }
+          })
+        ]);
+
+        if (!followAnonUser && !isUnfollowByUser) {
+          userFollowerAnonData.push({
+            follow_action_id: uuidv4(),
+            user_id_follower: followUser.user_id_follower,
+            user_id_followed: anonId,
+            is_anonymous: true
+          });
+
+          userFollowerAnonDataHistory.push({
+            user_id_follower: followUser.user_id_follower,
+            user_id_followed: anonId,
+            action: 'in',
+            source: `script`
+          });
+        }
+      })
+    );
+
+    let returnUserFollowUser = await UserFollowUser.bulkCreate(userFollowerAnonData);
+    await UserFollowUserHistory.bulkCreate(userFollowerAnonDataHistory);
+
+    return successResponse(res, 'Success Follow Anon Users', returnUserFollowUser);
+  } catch (error) {
+    console.log(error);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Avoid using `console.log` for error handling in production code. It's better to use a dedicated logging library that provides more flexibility and control over log levels, formats, and destinations.

```diff
-    console.log(error);
+    logger.error(error);
```
In the above snippet, `logger` is an instance of a logging library such as Winston or Bunyan.



<!-- This is an auto-generated comment by OSS CodeRabbit -->